app/src/main/java/org/olpc/xorobothash/MainActivity.java

```diff
@@ -0,0 +1,157 @@
+package org.olpc.xorobothash;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.util.Log;
+import android.widget.TextView;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.HttpURLConnection;
+
+public class MainActivity extends ActionBarActivity {
+
+    private ImageView robotImageView;
+
+    //For debugging
+    private static final String TAG = "XORobotHash";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        robotImageView = (ImageView) findViewById(R.id.robot_hash_image);
+        getRobot();
+    }
+
+    private void getRobot() {
```

Comment:
Style: the prefix `get` is most commonly used for field-access methods, i.e. just getting a field or easily computed value but having no side effects. This method does much more than that, and so I'd signal that by naming it `fetchRobot` or `refreshRobot` or something suggestive of more work.